src/Raven.Server/Documents/Indexes/Persistence/Corax/CoraxQueryBuilder.cs

```diff
@@ -253,7 +253,7 @@ internal static IQueryMatch BuildQuery(Parameters builderParameters, out OrderMe
                 coraxQuery = MaterializeWhenNeeded(builderParameters, coraxQuery, ref streamingOptimization);
             }
             // We sort on known field types, we'll optimize based on the first one to get the rest
-            else if (sortMetadata is [{ FieldType: MatchCompareFieldType.Floating or MatchCompareFieldType.Integer or MatchCompareFieldType.Sequence } sortBy, ..])
+            else if (sortMetadata is [{ FieldType: MatchCompareFieldType.Floating or MatchCompareFieldType.Integer or MatchCompareFieldType.Sequence, Field.FieldId: not CoraxConstants.IndexWriter.DynamicField } sortBy, ..])
```

Comment:
Please check https://github.com/ravendb/ravendb/pull/18988#issuecomment-2288833251 comment.

Reply:
@ayende For `include timings()`, do you mean adding message to `IncludeNonExistingMatch` node or to memoization that's used if we're sorting by dynamic field? In the second case I don't think we can pass the message from query builder to term provider

Reply:
Yes, some sort of note that we didn't use the optimal sorting method because on non existing.
Basically, so we'll have a way to figure out the reasons.

Also good for the future.

Reply:
I just added comment to code since we currently can't pass the message all the way from query builder to term provider